# Operators in Python

### - Arithmetic Operators
### - Relational Operators
### - Logical Operators
### - Bitwise Operators
### - Assignment Operators
### - Membership Operators

## 1) Arithmetic operators

In [2]:
# Addition
print(5 + 6)

# Subtraction
print(5 - 6)

# Multiplication
print(5 * 6)

# Division
print(5 / 2)

# Integer Division
print(5 // 2)

# Modules operator
print(5 % 2)

# Power of operator
print(5 ** 2)

11
-1
30
2.5
2
1
25


## 2) Relational operator

In [3]:
# Greater than operator
print(4 > 5)

# Less than operator
print(4 < 5)

# Greater than equal to operator
print(4 >= 4)

# Less than equal to operator
print(4 <= 4)

# Double equal to operator
print(4 == 4)

# Not equal to
print(4 != 4)

False
True
True
True
True
False


## 3) Logical Operator

In [4]:
# AND operator
print(1 and 0)

# OR operator
print(1 or 0)

# NOT operator
print(not 1)

0
1
False


## 4) Bitwise Operator

In [5]:
# bitwise and
print(2 & 3) # 10 & 11 ==> 10 means 2

# bitwise or
print(2 | 3) # 10 | 11 ==> 11 means 3

# bitwise xor
print(2 ^ 3) # 10 ^ 11 ==> 01 means 1

# bitwise not
print(~3) # 11 ==> 00 means -4

# Right shift operator
print(4 >> 2) # 100 >> 2 ==> 1 means 1

# Left shift operator
print(5 << 2) # 101 << 2 ==> 10100 means 20

2
3
1
-4
1
20


## 5) Assignment operators

In [6]:
# Equal to operator
# a = 2

a = 2

# a = a % 2
a %= 2

print(a)

# a++ ++a does not support Python

0


## 6) Membership operator

In [7]:
# in/not in (Case - sensitive)

print('D' in 'delhi')
print('D' in 'Delhi')

print('D' not in 'delhi')
print('D' not in 'Delhi')

print(1 in [1,2,3,4,5])

False
True
True
False
True


# Program
## Find the sum of a 3 digit number entered by the user

In [8]:
number = int(input('Enter a 3 digit number = '))

a = number % 10

number //= 10

b = number % 10

number //= 10

c = number

print('a :',a)
print('b :',b)
print('c :',c)

print('a+b+c :',a + b+ c)

Enter a 3 digit number = 345
a : 5
b : 4
c : 3
a+b+c : 12


# ********************************************************************************************

# If-else in Python

## login program and indentation

In [23]:
# email -> nitish.campusx@gmail.com
# password -> 1234

email = input('Enter email : ')
password = input('Enter password : ')

if email == '1234@gmail.com' and password == '1234':
    print('Welcome')
elif email == '1234@gmail.com' and password != '1234':
    print('Incorrect password')
    password = input('Enter password again : ')
    if password == '1234':
        print('Welcome, finally!')
    else:
        print('try again after sometime')
else:
    print('Invalid email')

Enter email : 1234@gmail.com
Enter password : 7867
Incorrect password
Enter password again : 1234
Welcome, finally!


## Find the Min of 3 numbers

In [1]:
a = int(input('first number = '))
b = int(input('second number = '))
c = int(input('third number = '))

if a < b and a < c :
    print('smallest is',a)
elif b < c :
    print('smallest is', b)
else :
    print('smallest is :', c)

first number = 4
second number = 5
third number = 3
smallest is : 3


## MENU DRIVEN CALCULATOR

In [2]:
fnum = int(input('Enter 1st number = '))
snum = int(input('Enter 2nd number = '))

op = input('Enter the opeartion = ')

if op == '+' :
    print('value :',fnum + snum)
elif op == '-' :
    print('value :',fnum - snum)
elif op == '*' :
    print('value :',fnum * snum)
elif op == '/' :
    print('value :',fnum / snum)

Enter 1st number = 12
Enter 2nd number = 3
Enter the opeartion = *
value : 36


## MENU DRIVEN PROGRAM

In [4]:
menu = input("""
Hi! how can I help you.
1. Enter 1 for pin change
2. Enter 2 for balance check
3. Enter 3 for withdrawl
4. Enter 4 for exit
""")

if menu == '1':
  print('Pin change')
elif menu == '2':
  print('Balance')
elif menu == '3' :
    print('Withdraw')
else:
  print('Exit')


Hi! how can I help you.
1. Enter 1 for pin change
2. Enter 2 for balance check
3. Enter 3 for withdrawl
4. Enter 4 for exit
4
Exit


# ********************************************************************************************

# Modules in Python

### - math
### - keywords
### - random
### - datetime

## 1) math

In [5]:
import math

math.factorial(5)
math.floor(6.8)
math.sqrt(196)

14.0

## 2) keywords

In [7]:
import keyword

print(keyword.kwlist)

['False', 'None', 'True', 'and', 'as', 'assert', 'async', 'await', 'break', 'class', 'continue', 'def', 'del', 'elif', 'else', 'except', 'finally', 'for', 'from', 'global', 'if', 'import', 'in', 'is', 'lambda', 'nonlocal', 'not', 'or', 'pass', 'raise', 'return', 'try', 'while', 'with', 'yield']


## 3) random

In [8]:
import random

print(random.randint(1,100))

27


## 4) datetime

In [9]:
import datetime

print(datetime.datetime.now)

<built-in method now of type object at 0x00007FFDB7CD4FB0>


In [10]:
help('modules')


Please wait a moment while I gather a list of all available modules...



C:\Users\yashg\anaconda3\lib\site-packages\caffe2\__init__.py:5: UserWarning: Caffe2 support is not fully enabled in this PyTorch build. Please enable Caffe2 by building PyTorch from source with `BUILD_CAFFE2=1` flag.
  warnings.warn("Caffe2 support is not fully enabled in this PyTorch build. "
C:\Users\yashg\anaconda3\lib\site-packages\caffe2\proto\__init__.py:17: UserWarning: Caffe2 support is not enabled in this PyTorch build. Please enable Caffe2 by building PyTorch from source with `BUILD_CAFFE2=1` flag.
  warnings.warn('Caffe2 support is not enabled in this PyTorch build. '
C:\Users\yashg\anaconda3\lib\site-packages\caffe2\proto\__init__.py:17: UserWarning: Caffe2 support is not enabled in this PyTorch build. Please enable Caffe2 by building PyTorch from source with `BUILD_CAFFE2=1` flag.
  warnings.warn('Caffe2 support is not enabled in this PyTorch build. '
C:\Users\yashg\anaconda3\lib\site-packages\caffe2\python\__init__.py:9: UserWarning: Caffe2 support is not enabled in this

IPython             conda_build         mmapfile            socketserver
OpenSSL             conda_content_trust mmsystem            socks
PIL                 conda_env           mock                sockshandler
PyQt5               conda_pack          modulefinder        sortedcontainers
TBB                 conda_package_handling mpmath              soupsieve
__future__          conda_package_streaming msgpack             sphinx
_abc                conda_token         msilib              spyder
_aix_support        conda_verify        msvcrt              spyder_kernels
_argon2_cffi_bindings configparser        multipledispatch    sqlalchemy
_ast                constantly          multiprocessing     sqlite3
_asyncio            contextlib          munkres             sre_compile
_bisect             contextvars         mypy_extensions     sre_constants
_black_version      contourpy           nacl                sre_parse
_blake2             cookiecutter        navigator_updater   ssl
_boo

# ***********************************************************************************************

# LOOPS

### Loops in python is used to iterate over a sequence (list,tuple,set,dictionary, and string)

## 1) While loop

## Program
### Print a table of a number given by user

In [12]:
number = int(input('Enter the number of a table = '))

i = 1
while i < 11 :
    print(number,'*',i,'=',number * i)
    i += 1

Enter the number of a table = 12
12 * 1 = 12
12 * 2 = 24
12 * 3 = 36
12 * 4 = 48
12 * 5 = 60
12 * 6 = 72
12 * 7 = 84
12 * 8 = 96
12 * 9 = 108
12 * 10 = 120


## While loops with else

In [14]:
x = 1

while x < 3 :
    print(x)
    x += 1
    
else :
    print('limit crossed')

1
2
limit crossed


## Program
### Guessing Game

In [15]:
# generate a random number between 1 and 100

import random

jackpot = random.randint(1,100)

guess = int(input('guess a number = '))

attempts = 1
while guess != jackpot:
    if guess < jackpot :
      print('Wrong! guess higher')
    else :
      print('Wrong! guess lower')
    
    guess = int(input('guess a number = '))
    attempts += 1
    
else:
    print('correct guess')
    print('attempts :',attempts)

guess a number = 54
Wrong! guess higher
guess a number = 70
Wrong! guess higher
guess a number = 80
Wrong! guess higher
guess a number = 92
Wrong! guess lower
guess a number = 85
Wrong! guess higher
guess a number = 88
Wrong! guess lower
guess a number = 86
Wrong! guess higher
guess a number = 87
correct guess
attempts : 8


## 2) For loop

In [16]:
for i in range(1,7):
    print(i)

1
2
3
4
5
6


In [17]:
for i in range(1,11,3):
    print(i)

1
4
7
10


In [18]:
for i in range(7,0,-1):
    print(i)

7
6
5
4
3
2
1


In [19]:
for i in range(7,0,-2):
    print(i)

7
5
3
1


In [20]:
for i in 'Kanina':
    print(i)

K
a
n
i
n
a


In [21]:
for i in (1,2,3,4,5):
    print(i)

1
2
3
4
5


In [22]:
for i in {1,2,3,4,5}:
    print(i)

1
2
3
4
5


In [23]:
for i in [1,2,3,4,5]:
    print(i)

1
2
3
4
5


# Program
##  The current population of a town is 10000. The population of the town is increasing at the rate of 10% per year. You have to write a program to find out the population at the end of each of the last 10 years.

In [1]:
curr_pop = 10000

for i in range(10,0,-1):
    print(i,int(curr_pop))
    curr_pop = curr_pop / 1.1

10 10000
9 9090
8 8264
7 7513
6 6830
5 6209
4 5644
3 5131
2 4665
1 4240


# Program
## Sequence Sum<br>1/1! + 2/2! + 3/3! + ...

In [3]:
n = int(input('enter n = '))

result = 0
fact = 1

for i in range(1,n+1) :
    fact = fact * i
    result = result + i/fact
    
    print(round(result,6))

enter n = 10
1.0
2.0
2.5
2.666667
2.708333
2.716667
2.718056
2.718254
2.718279
2.718282


# 3) NESTED LOOPS

In [4]:
# Unique pairs of two-digits from 1 to 4 numbers

for i in range(1,5) :
    for j in range(1,5) :
        print(i,j)

1 1
1 2
1 3
1 4
2 1
2 2
2 3
2 4
3 1
3 2
3 3
3 4
4 1
4 2
4 3
4 4


## Pattern 1

In [5]:
rows = int(input('enet number of rows = '))

for i in range(1,rows + 1) :
    for j in range(1,i+1) :
        print('*',end='')
    print()

enet number of rows = 5
*
**
***
****
*****


## Pattern 2

In [6]:
rows = int(input('enter number of rows = '))

for i in range(1,rows+1) :
    for j in range(1,i+1) :
        print(j,end='')
    for k in range(j-1,0,-1) :
        print(k,end='')
    print()

enter number of rows = 5
1
121
12321
1234321
123454321


# Loop Control Statement
## - Break
## - Continue
## - Pass

## i) Break

In [7]:
for i in range(1,10) :
    if i == 5:
        break
    print(i)

1
2
3
4


# Program
## Print all the prime numbers in between Lower & Upper range

In [8]:
lower = int(input('enter lower range = '))
upper = int(input('enter upper range = '))

for i in range(lower,upper+1) :
    for j in range(2,i) :
        if i % j == 0 :
            break
    else :
        print(i)

enter lower range = 10
enter upper range = 100
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97


## ii) Continue

In [9]:
for i in range(1,10) :
    if i == 5:
        continue
    print(i)

1
2
3
4
6
7
8
9


## iii) pass

In [10]:
for i in range(1,10):
  pass

# ********************************************************************************************

# Task 1
## Write a program that will give you in hand monthly salary after deduction on CTC - HRA(10%), DA(5%), PF(3%) and taxes deduction as below:
### Salary(Lakhs) : Tax(%)

### Below 5 : 0%<br>5-10 : 10%<br>10-20 : 20%<br>aboove 20 : 30%

In [1]:
ctc = int(input('Enter your annual CTC = '))

if ctc < 500000:
    salary = ctc * .82
elif ctc < 1000000:
    salary = ctc * .72
elif ctc < 2000000:
    salary = ctc * .62
elif ctc < 3000000:
    salary = ctc * .52
    
print("You in hand monthly salary will be :",round(salary/12,2))

Enter your annual CTC = 800000
You in hand monthly salary will be : 48000.0


# Task 2
## Write a program that take a user input of three angles and will find out whether it can form a triangle or not.<br>Hint - Sum of all angles is 180 and all angles are positive

In [2]:
first = int(input('enter the 1st angle = '))
second = int(input('enter the 2nd angle = '))
third = int(input('enter the 3rd angle = '))

if (first + second + third) == 180 and first > 0 and second > 0 and third > 0:
    print('forms a triangle')
else:
  print('does not form a triangle')

enter the 1st angle = 70
enter the 2nd angle = 50
enter the 3rd angle = 60
forms a triangle


# Task 3
## Write a program that will take user input of cost price and selling price and determines whether its a loss or a profit.

In [3]:
cp = int(input('Enter cost price = '))
sp = int(input('Enter selling price = '))

if cp > sp :
    print('Loss')
elif cp < sp :
    print('Profit')
else :
    print('No Profit No Loss')

Enter cost price = 200
Enter selling price = 240
Profit


# Task 4
## Write a menu-driven program -
###       1) cm to ft
### 2) km to miles
### 3) USD to INR
### 4) exit

## Hint :
### - 1 cm = 0.032ft
### - 1km = 0.62 miles
### - 1 USD = 80 INR

In [4]:
menu = input("""
Hi select an option
1. cms to ft
2. km to miles
3. USD to INR
4. Exit
""")

if menu == '1' :
    cm = float(input('Enter cm value = '))
    print('ft value is :',cm*0.032)
if menu == '2' :
    km = float(input('Enter km value = '))
    print('miles value is :',km*0.62)
if menu == '3' :
    usd = float(input('Enter usd value = '))
    print('INR value is :',usd*80)
else :
    exit()


Hi select an option
1. cms to ft
2. km to miles
3. USD to INR
4. Exit
3
Enter usd value = 20
INR value is : 1600.0


# Task 5
## Display Fibonacci series up to 10 terms.

In [5]:
num1, num2 = 0,1

for i in range(10) : # range is 0 to 9
    print(num1)
    
    next = num1 + num2
    
    num1 = num2
    num2 = next

0
1
1
2
3
5
8
13
21
34


# Task 6
##  Find the factorial of a given number.

In [6]:
num = int(input('enter the number'))

fact = 1
for i in range(1,num+1) :
    fact = fact*i
    
    print(fact)

enter the number5
1
2
6
24
120


# Task 7
## Reverse an integer number

In [7]:
number = int(input('Enter a number = '))

rev = 0

while number > 0 :
    last = number % 10
    rev = rev * 10 + last
    number //= 10
    
print('Reverse no is :',rev)

Enter a number = 12345
Reverse no is : 54321


# Task 8
## Take a user input as integer N. Find out the sum from 1 to N. If any number if divisible by 5, then skip that number. And if the sum is greater than 300, don't need to calculate the sum further more. Print the final result. And don't use for loop to solve this problem.

In [12]:
# using for loop

n = int(input('enter a number = '))

sum = 0
for i in range(1,n+1) :
    if i % 5 == 0 :
        continue
    sum += i
    
    if sum > 300 :
        sum -= i
        break
print(sum)

enter a number = 30
276


In [13]:
# using while loop

n = int(input('enter a number = '))

sum = 0
i = 1

while i <= n :
    if i % 5 == 0 :
        i += 1
        continue
    sum += i
    
    if sum > 300 :
        sum -= i
        break
    i += 1
print(sum)

enter a number = 30
276


# Task 9
## Write a program that keeps on accepting a number from the user until the user enters Zero. Display the sum and average of all the numbers.

In [14]:
sum = 0
count = 0

while True:
    num = int(input('enter number = '))
    if num == 0:
        break
    sum += num
    count += 1
    
print('sum :',sum)
print('avg :',sum/count)

enter number = 10
enter number = 20
enter number = 30
enter number = 0
sum : 60
avg : 20.0


# Task 10
## Write a program which will find all such numbers which are divisible by 7 but are not a multiple of 5, between 2000 and 3200 (both included). The numbers obtained should be printed in a comma-separated sequence on a single line.

In [15]:
L = []
for i in range(2000,3201) :
    if i % 7 == 0 and i % 5 != 0 :
        L.append(str(i))
        
print(','.join(L))

2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996,3003,3017,3024,3031,3038,3052,3059,3066,3073,3087,3094,3101,3108,3122,3129,3136,3143,3157,3164,3171,3178,3192,3199


# Task 11
## Write a program, which will find all such numbers between 1000 and 3000 (both included) such that each digit of the number is an even number. The numbers obtained should be printed in a space-separated sequence on a single line.

In [19]:
L = []

for i in range(1000,3001) :
    flag = True
    
    curr = i
    
    while curr > 0 :
        last = curr % 10
        if last % 2 != 0 :
            flag = False
            break
        curr //= 10
    
    if flag == True:
        L.append(str(i))
        
print(','.join(L))

2000,2002,2004,2006,2008,2020,2022,2024,2026,2028,2040,2042,2044,2046,2048,2060,2062,2064,2066,2068,2080,2082,2084,2086,2088,2200,2202,2204,2206,2208,2220,2222,2224,2226,2228,2240,2242,2244,2246,2248,2260,2262,2264,2266,2268,2280,2282,2284,2286,2288,2400,2402,2404,2406,2408,2420,2422,2424,2426,2428,2440,2442,2444,2446,2448,2460,2462,2464,2466,2468,2480,2482,2484,2486,2488,2600,2602,2604,2606,2608,2620,2622,2624,2626,2628,2640,2642,2644,2646,2648,2660,2662,2664,2666,2668,2680,2682,2684,2686,2688,2800,2802,2804,2806,2808,2820,2822,2824,2826,2828,2840,2842,2844,2846,2848,2860,2862,2864,2866,2868,2880,2882,2884,2886,2888


# Task 12
## A robot moves in a plane starting from the original point (0,0). The robot can move toward UP, DOWN, LEFT and RIGHT with a given steps.
### The trace of robot movement is shown as the following:
### UP 5
### DOWN 3
### LEFT 3
### RIGHT 2
### !
### ! means robot stop there.
## Write a program to compute the distance from current position after a sequence of movement and original point.

### If the distance is a float, then just print the nearest integer.

In [25]:
pos = [0,0]
while True:
    s = input('enter a robot path = ')
    if s == '!' :
        break
    direction = s.split()[0]
    steps = int(s.split()[1])
    
    if direction == 'LEFT':
        pos[0] = pos[0] - steps
    elif direction == 'RIGHT':
        pos[0] = pos[0] + steps
    elif direction == 'UP':
        pos[1] = pos[1] + steps
    elif direction == 'DOWN':
        pos[1] = pos[1] - steps
    else :
        pass

print('new pos',pos)
print(int((pos[0]**2 + pos[1]**2)**0.5))

enter a robot path = UP 5
enter a robot path = RIGHT 7
enter a robot path = LEFT 9
enter a robot path = DOWN 1
enter a robot path = !
new pos [-2, 4]
4


# Task 13
## Write a program to print whether a given number is a prime number or not

In [1]:
num = int(input('Enter the number = '))

flag = True
for i in range(2,num) :
    if num % i == 0:
        flag = False
        break
        
if flag == True:
    print('Prime')
else :
    print('Not prime')

Enter the number = 13
Prime


# Task 14
## Print the following pattern. Write a program to use for loop to print the following reverse number pattern.
5 4 3 2 1

4 3 2 1

3 2 1

2 1

1

In [2]:
rows = int(input('enter the rows = '))

for i in range(0,rows):
    for j in range(rows-i,0,-1):
        print(j,end=' ')
    print()

enter the rows = 5
5 4 3 2 1 
4 3 2 1 
3 2 1 
2 1 
1 


# Task 15
## Print the following pattern
```bash
* 
* * 
* * * 
* * * * 
* * * * * 
* * * * 
* * * 
* * 
*
```

In [4]:
rows = int(input('enter the rows = '))

for i in range(1,rows+1):
    for j in range(1,i+1):
        print('*',end=' ')
    print()
    
for i in range(1,rows):
    for j in range(rows-i,0,-1):
        print('*',end=' ')
    print()

enter the rows = 5
* 
* * 
* * * 
* * * * 
* * * * * 
* * * * 
* * * 
* * 
* 


# Task 16
## Write a program to pring the following pattern
```bash
   *
  * *
 * * *
* * * *
```

In [5]:
rows = int(input('enter the rows'))

for i in range(1,rows+1):
    print(' '*(rows-1),end='')
    print('* '*i)
    rows -= 1

enter the rows4
   * 
  * * 
 * * * 
* * * * 


# Task 17
## Write a program to print the following pattern

1 

2 1 

3 2 1 

4 3 2 1 

5 4 3 2 1

In [6]:
rows = int(input('enter the rows'))

for i in range(1,rows+1):
    for j in range(i,0,-1):
        print(j,end=' ')
    print()

enter the rows5
1 
2 1 
3 2 1 
4 3 2 1 
5 4 3 2 1 


# Task 18
## Write a Python Program to Find the Sum of the Series till the nth term:<br> 
### 1 + x^2/2 + x^3/3 + … x^n/n<br>n will be provided by the user

In [8]:
x = int(input('enter x'))
n = int(input('enter nth term'))

sum = 1
s = ''
for i in range(2,n+1):
    sum += x**i/i
    s = s + 'x^{}/{} +'.format(i,i)
print('1 +',s[:-1])
print(sum)

enter x10
enter nth term5
1 + x^2/2 +x^3/3 +x^4/4 +x^5/5 
22884.333333333332


# Task 19
## (1/1)*(x-1/x)^1 + (1/2)*(x-1/x)^2 + (1/3)*(x-1/x)^3 + (1/n)*(x-1/x)^n

In [9]:
x = 10
n = 5

sum = 0
s = ''
for i in range(1,n+1):
    sum += (1/i)*((x-1)/x)**i
    s = s + '(1/{})*((x-1)/x)**{} +'.format(i,i)
    
print(s[:-1])

print(sum)

(1/1)*((x-1)/x)**1 +(1/2)*((x-1)/x)**2 +(1/3)*((x-1)/x)**3 +(1/4)*((x-1)/x)**4 +(1/5)*((x-1)/x)**5 
1.8301230000000004


# Task 20
## Find the sum of the series upto n terms.
## Write a program to calculate the sum of series up to n term. For example, if n =5 the series will become 2 + 22 + 222 + 2222 + 22222 = 24690. Take the user input and then calculate. And the output style should match which is given in the example.

In [10]:
n = int(input('enter the number of terms = '))

start = 2
sum = 0
for i in range(0,n):
    if i < n-1:
        print(start,end='+')
    else:
        print(start)
    sum = sum + start
    start = start * 10 + 2
    
print(sum)

enter the number of terms = 5
2+22+222+2222+22222
24690


# Task 21
## Write a program to print all the unique combinations of 1,2,3 and 4
`Output`:
```
1 2 3 4
1 2 4 3
1 3 2 4
1 3 4 2
1 4 2 3
1 4 3 2
2 1 3 4
2 1 4 3
2 3 1 4
2 3 4 1
2 4 1 3
.
.
and so on
```

In [11]:
for i in range(1,5):
    for j in range(1,5):
        for k in range(1,5):
            for m in range(1,5):
                print(i,j,k,m)

1 1 1 1
1 1 1 2
1 1 1 3
1 1 1 4
1 1 2 1
1 1 2 2
1 1 2 3
1 1 2 4
1 1 3 1
1 1 3 2
1 1 3 3
1 1 3 4
1 1 4 1
1 1 4 2
1 1 4 3
1 1 4 4
1 2 1 1
1 2 1 2
1 2 1 3
1 2 1 4
1 2 2 1
1 2 2 2
1 2 2 3
1 2 2 4
1 2 3 1
1 2 3 2
1 2 3 3
1 2 3 4
1 2 4 1
1 2 4 2
1 2 4 3
1 2 4 4
1 3 1 1
1 3 1 2
1 3 1 3
1 3 1 4
1 3 2 1
1 3 2 2
1 3 2 3
1 3 2 4
1 3 3 1
1 3 3 2
1 3 3 3
1 3 3 4
1 3 4 1
1 3 4 2
1 3 4 3
1 3 4 4
1 4 1 1
1 4 1 2
1 4 1 3
1 4 1 4
1 4 2 1
1 4 2 2
1 4 2 3
1 4 2 4
1 4 3 1
1 4 3 2
1 4 3 3
1 4 3 4
1 4 4 1
1 4 4 2
1 4 4 3
1 4 4 4
2 1 1 1
2 1 1 2
2 1 1 3
2 1 1 4
2 1 2 1
2 1 2 2
2 1 2 3
2 1 2 4
2 1 3 1
2 1 3 2
2 1 3 3
2 1 3 4
2 1 4 1
2 1 4 2
2 1 4 3
2 1 4 4
2 2 1 1
2 2 1 2
2 2 1 3
2 2 1 4
2 2 2 1
2 2 2 2
2 2 2 3
2 2 2 4
2 2 3 1
2 2 3 2
2 2 3 3
2 2 3 4
2 2 4 1
2 2 4 2
2 2 4 3
2 2 4 4
2 3 1 1
2 3 1 2
2 3 1 3
2 3 1 4
2 3 2 1
2 3 2 2
2 3 2 3
2 3 2 4
2 3 3 1
2 3 3 2
2 3 3 3
2 3 3 4
2 3 4 1
2 3 4 2
2 3 4 3
2 3 4 4
2 4 1 1
2 4 1 2
2 4 1 3
2 4 1 4
2 4 2 1
2 4 2 2
2 4 2 3
2 4 2 4
2 4 3 1
2 4 3 2
2 4 3 3
2 4 3 4
2 4 4 1


# Task 24
## Given a string create short form ofthe string from Initial character. Short form should be capitalised.

Example:

`Input:`
```
Lage Raho Munna Bhai
```

`Output:`
```
LRMB
```

In [3]:
inp = 'Lage Raho Munna Bhai'
res = ''

for i in inp.split():
    res = res + i[0].upper()

print(res)

LRMB


## Task 25
# Append second string in the middle of first string

`Input:`
```
innovative
data
```
`Output`:

```
innovdataative
```

In [5]:
s1 = input('enter the 1st string = ')
s2 = input('enter the 2nd string = ')

print(s1[0:len(s1)//2] + s2 + s1[len(s1)//2:])

enter the 1st string = innovative
enter the 2nd string = data
innovdataative


# Task 26
## Given string contains a combination of the lower and upper case letters. Write a program to arrange the characters of a string so that all lowercase letters should come first.

Given:

`str1 = PyNaTive`

Expected Output:

`yaivePNT`

In [6]:
s = 'PyNaTive'

upper = ''
lower = ''

for i in s:
    if i.islower():
        lower += i
    upper += i
    
print(lower + upper)

yaivePyNaTive


# Task 27
## Take a alphanumeric string input and print the sum and average of the digits that appear in the string, ignoring all other characters.

Input:

`hel123O4every093`

Output:

```
Sum: 22
Avg: 3.14
```

In [7]:
s = 'hel123O4every093'

sum = 0
count = 0

for i in s:
    if i.isdigit():
        sum = sum + int(i)
        count += 1
        
print(sum/count)

3.142857142857143


# Task 28
## Removal of all characters from a string except integers
`Given:`

```
str1 = 'I am 25 years and 10 months old'
```
`Expected Output:`
```
2510
```

In [8]:
s = 'I am 25 years and 10 months old'
res = ''

for i in s:
    if i.isdigit():
        res += i
        
print(res)

2510
